libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -575,7 +575,7 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 			return true;
 		}
 	);
-	solAssert(errors.empty(), "Code generation for inline assembly with errors requested.");
+	solAssert(errors.empty() || Error::containsOnlyWarnings(errors), "Code generation for inline assembly with errors requested.");
```

Comment:
Yes. The only part I disliked about `containsOnlyWarnings` that is shadows the fact if tthere are no warnings/errors at all. However that is not a problem here.